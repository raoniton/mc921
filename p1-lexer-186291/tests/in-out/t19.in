// Base class
class Animal {
    int age;
    String name;

    // Method
    public void makeSound() {
        print("Animal sound");
    }
}

// Derived class
class Dog extends Animal {
    String breed;

    // Override method
    public void makeSound() {
        print("Woof!");
    }

    // New method
    public void fetch() {
        print(name, " is fetching.");
    }
}

// Main class
class Main {
    public static void main(String[] args) {
        // Create an instance of Animal
        Animal myAnimal = new Animal();
        myAnimal.age = 5;
        myAnimal.name = "Generic Animal";
        myAnimal.makeSound(); // Output: Animal sound

        // Create an instance of Dog
        Dog myDog = new Dog();
        myDog.age = 3;
        myDog.name = "Buddy";
        myDog.breed = "Golden Retriever";
        myDog.makeSound(); // Output: Woof!
        myDog.fetch();     // Output: Buddy is fetching.

        // Polymorphism: Animal reference to Dog object
        Animal anotherAnimal = new Dog();
        anotherAnimal.age = 4;
        anotherAnimal.name = "Max";
        anotherAnimal.makeSound(); // Output: Woof!
    }
}