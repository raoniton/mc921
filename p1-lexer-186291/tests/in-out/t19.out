LexToken(CLASS,'class',2,14)
LexToken(ID,'Animal',2,20)
LexToken(LBRACE,'{',2,27)
LexToken(INT,'int',3,33)
LexToken(ID,'age',3,37)
LexToken(SEMI,';',3,40)
LexToken(STRING,'String',4,46)
LexToken(ID,'name',4,53)
LexToken(SEMI,';',4,57)
LexToken(PUBLIC,'public',7,78)
LexToken(VOID,'void',7,85)
LexToken(ID,'makeSound',7,90)
LexToken(LPAREN,'(',7,99)
LexToken(RPAREN,')',7,100)
LexToken(LBRACE,'{',7,102)
LexToken(PRINT,'print',8,112)
LexToken(LPAREN,'(',8,117)
LexToken(STRING_LITERAL,'"Animal sound"',8,118)
LexToken(RPAREN,')',8,132)
LexToken(SEMI,';',8,133)
LexToken(RBRACE,'}',9,139)
LexToken(RBRACE,'}',10,141)
LexToken(CLASS,'class',13,161)
LexToken(ID,'Dog',13,167)
LexToken(EXTENDS,'extends',13,171)
LexToken(ID,'Animal',13,179)
LexToken(LBRACE,'{',13,186)
LexToken(STRING,'String',14,192)
LexToken(ID,'breed',14,199)
LexToken(SEMI,';',14,204)
LexToken(PUBLIC,'public',17,234)
LexToken(VOID,'void',17,241)
LexToken(ID,'makeSound',17,246)
LexToken(LPAREN,'(',17,255)
LexToken(RPAREN,')',17,256)
LexToken(LBRACE,'{',17,258)
LexToken(PRINT,'print',18,268)
LexToken(LPAREN,'(',18,273)
LexToken(STRING_LITERAL,'"Woof!"',18,274)
LexToken(RPAREN,')',18,281)
LexToken(SEMI,';',18,282)
LexToken(RBRACE,'}',19,288)
LexToken(PUBLIC,'public',22,313)
LexToken(VOID,'void',22,320)
LexToken(ID,'fetch',22,325)
LexToken(LPAREN,'(',22,330)
LexToken(RPAREN,')',22,331)
LexToken(LBRACE,'{',22,333)
LexToken(PRINT,'print',23,343)
LexToken(LPAREN,'(',23,348)
LexToken(ID,'name',23,349)
LexToken(COMMA,',',23,353)
LexToken(STRING_LITERAL,'" is fetching."',23,355)
LexToken(RPAREN,')',23,370)
LexToken(SEMI,';',23,371)
LexToken(RBRACE,'}',24,377)
LexToken(RBRACE,'}',25,379)
LexToken(CLASS,'class',28,396)
LexToken(ID,'Main',28,402)
LexToken(LBRACE,'{',28,407)
LexToken(PUBLIC,'public',29,413)
LexToken(STATIC,'static',29,420)
LexToken(VOID,'void',29,427)
LexToken(MAIN,'main',29,432)
LexToken(LPAREN,'(',29,436)
LexToken(STRING,'String',29,437)
LexToken(LBRACKET,'[',29,443)
LexToken(RBRACKET,']',29,444)
LexToken(ID,'args',29,446)
LexToken(RPAREN,')',29,450)
LexToken(LBRACE,'{',29,452)
LexToken(ID,'Animal',31,502)
LexToken(ID,'myAnimal',31,509)
LexToken(ASSIGN,'=',31,518)
LexToken(NEW,'new',31,520)
LexToken(ID,'Animal',31,524)
LexToken(LPAREN,'(',31,530)
LexToken(RPAREN,')',31,531)
LexToken(SEMI,';',31,532)
LexToken(ID,'myAnimal',32,542)
LexToken(DOT,'.',32,550)
LexToken(ID,'age',32,551)
LexToken(ASSIGN,'=',32,555)
LexToken(INT_LITERAL,'5',32,557)
LexToken(SEMI,';',32,558)
LexToken(ID,'myAnimal',33,568)
LexToken(DOT,'.',33,576)
LexToken(ID,'name',33,577)
LexToken(ASSIGN,'=',33,582)
LexToken(STRING_LITERAL,'"Generic Animal"',33,584)
LexToken(SEMI,';',33,600)
LexToken(ID,'myAnimal',34,610)
LexToken(DOT,'.',34,618)
LexToken(ID,'makeSound',34,619)
LexToken(LPAREN,'(',34,628)
LexToken(RPAREN,')',34,629)
LexToken(SEMI,';',34,630)
LexToken(ID,'Dog',37,702)
LexToken(ID,'myDog',37,706)
LexToken(ASSIGN,'=',37,712)
LexToken(NEW,'new',37,714)
LexToken(ID,'Dog',37,718)
LexToken(LPAREN,'(',37,721)
LexToken(RPAREN,')',37,722)
LexToken(SEMI,';',37,723)
LexToken(ID,'myDog',38,733)
LexToken(DOT,'.',38,738)
LexToken(ID,'age',38,739)
LexToken(ASSIGN,'=',38,743)
LexToken(INT_LITERAL,'3',38,745)
LexToken(SEMI,';',38,746)
LexToken(ID,'myDog',39,756)
LexToken(DOT,'.',39,761)
LexToken(ID,'name',39,762)
LexToken(ASSIGN,'=',39,767)
LexToken(STRING_LITERAL,'"Buddy"',39,769)
LexToken(SEMI,';',39,776)
LexToken(ID,'myDog',40,786)
LexToken(DOT,'.',40,791)
LexToken(ID,'breed',40,792)
LexToken(ASSIGN,'=',40,798)
LexToken(STRING_LITERAL,'"Golden Retriever"',40,800)
LexToken(SEMI,';',40,818)
LexToken(ID,'myDog',41,828)
LexToken(DOT,'.',41,833)
LexToken(ID,'makeSound',41,834)
LexToken(LPAREN,'(',41,843)
LexToken(RPAREN,')',41,844)
LexToken(SEMI,';',41,845)
LexToken(ID,'myDog',42,872)
LexToken(DOT,'.',42,877)
LexToken(ID,'fetch',42,878)
LexToken(LPAREN,'(',42,883)
LexToken(RPAREN,')',42,884)
LexToken(SEMI,';',42,885)
LexToken(ID,'Animal',45,986)
LexToken(ID,'anotherAnimal',45,993)
LexToken(ASSIGN,'=',45,1007)
LexToken(NEW,'new',45,1009)
LexToken(ID,'Dog',45,1013)
LexToken(LPAREN,'(',45,1016)
LexToken(RPAREN,')',45,1017)
LexToken(SEMI,';',45,1018)
LexToken(ID,'anotherAnimal',46,1028)
LexToken(DOT,'.',46,1041)
LexToken(ID,'age',46,1042)
LexToken(ASSIGN,'=',46,1046)
LexToken(INT_LITERAL,'4',46,1048)
LexToken(SEMI,';',46,1049)
LexToken(ID,'anotherAnimal',47,1059)
LexToken(DOT,'.',47,1072)
LexToken(ID,'name',47,1073)
LexToken(ASSIGN,'=',47,1078)
LexToken(STRING_LITERAL,'"Max"',47,1080)
LexToken(SEMI,';',47,1085)
LexToken(ID,'anotherAnimal',48,1095)
LexToken(DOT,'.',48,1108)
LexToken(ID,'makeSound',48,1109)
LexToken(LPAREN,'(',48,1118)
LexToken(RPAREN,')',48,1119)
LexToken(SEMI,';',48,1120)
LexToken(RBRACE,'}',49,1143)
LexToken(RBRACE,'}',50,1145)
